images/capi/ansible/roles/providers/tasks/main.yml

```
@@ -112,14 +112,23 @@
     mode: "0644"
   when: ansible_os_family == "Debian"
 
-- name: Set cloudinit feature flags for redhat
+- name: Set cloudinit feature flags for redhat 8
   copy:
     src: usr/lib/python3/site-packages/cloudinit/feature_overrides.py
     dest: /usr/lib/python3.6/site-packages/cloudinit/feature_overrides.py
     owner: root
     group: root
     mode: "0644"
-  when: ansible_os_family == "RedHat" and ansible_distribution == "RedHat"
+  when: ansible_os_family == "RedHat" and ansible_distribution == "RedHat" and ansible_distribution_major_version == "8"
+
+- name: set cloudinit feature flags for redhat 9
```

```suggestion
- name: Set cloudinit feature flags for redhat 9
```

This broke an `ansible-lint` rule apparently:

```
WARNING  Listing 1 violation(s) that are fatal
name[casing]: All names should start with an uppercase letter.
ansible/roles/providers/tasks/main.yml:124 Task/Handler: set cloudinit feature flags for redhat 9
```